Assets/Scripts/AI/Cartographer.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace AI
+{
+    public class Cartographer
+    {
+        private readonly List<AiWrapper> _gameObjects = new List<AiWrapper>();
+        private readonly Map _map;
+
+        /// <summary>
+        /// Constructs a cartographer for a given width and height
+        /// </summary>
+        /// <param name="width">Width of the map.</param>
+        /// <param name="height">Height of the map.</param>
+        public Cartographer(int width, int height)
+        {
+            _map = new Map(width, height);
+            FillStaticMap();
+        }
+
+        /// <summary>
+        /// Converts the matrix to a array which is readable by neuronal networks.
+        /// </summary>
+        /// <returns>float matrix [x,y, information]</returns>
+        public float[,] MatrixNnReady()
+        {
+            int infoDimension = 0;
+
+            List<MapEntry> mapEntries = _map.ToList();
+            foreach (MapEntry entry in mapEntries.Where(entry => entry.Dimension() > infoDimension))
+            {
+                infoDimension = entry.Dimension();
+            }
```

Comment:
isnt there a builtin Max function for LINQ?

Reply:
yes there is.